dra-evolution/pkg/api/capacity_types.go

```diff
@@ -154,40 +184,24 @@ type DeviceAttribute struct {
 	VersionValue *string `json:"version,omitempty" protobuf:"bytes,5,opt,name=version"`
 }
 
-type SharedCapacity struct {
-	// Name is a unique identifier among all shared capacities managed by the
+type DeviceCapacity struct {
+	// Name is a unique identifier among all capacities managed by the
 	// driver in the pool.
 	//
-	// It is referenced both when defining the total amount of shared capacity
-	// that is available, as well as by individual devices when declaring
-	// how much of this shared capacity they consume.
-	//
-	// SharedCapacity names must be a C-style identifier (e.g. "the_name") with
-	// a maximum length of 32.
-	//
-	// By limiting these names to a C-style identifier, the same validation can
-	// be used for both these names and the identifier portion of a
-	// DeviceAttribute name.
-	//
 	// +required
 	Name string `json:"name"`
 
 	// Capacity is the total capacity of the named resource.
-	// This can either represent the total *available* capacity, or the total
-	// capacity *consumed*, depending on the context where it is referenced.
 	//
 	// +required
 	Capacity resource.Quantity `json:"capacity"`
```

Comment:
I would make this. one-of for local capacity (now) and shared capacity (later)

Reply:
In this version, that is handled by using a the `CommonDeviceData` one-of (`Capacity` and `SharedCapacity` in that one-of). We could also add a separate, `SharedCapacity` field at the device level if we wanted individual devices to offer shared capacity (depends on how we end up doing partitions whether that makes any sense or not).

Reply:
I don't know that we want to model it exactly that way, and it won't come into play until we have a different discriminator value, so I don't think it's necessary.

Reply:
Sure. If we are not in agreement, then what is there now seems fine. As you said, we can have a separate SharedCapacity slice in the future to cover this.